### ISSUE TITLE
Builds with broken allow_failures show as green on github

### ISSUE THREAD

**Issue by gnzlbg** [travis-ci/travis-ci#8716]
I would like to easily recognize when a build that is allowed to fail fails. Currently, when this is the case, the build shows as "green", all tests pass.

I'd like to have the option to make the builds show up as "orange", as in, only the allowed_failure builds fail. Maybe one could indicate this by some flag on the travis.yml file, e.g., by using `warn_on_failure` instead of `allow_failure`.

**Comment by adam-moss** [comment]
I second this one, the difference between "passed" and "pass with warnings" can be quite important.

In my particular use case I use this with things like `npm outdated`.  If there are outdated modules I don't want the build to fail (so the job is included in `allow_failures`) but I do want the overall build status to reflect that all the steps didn't pass.

**Comment by tlyu** [comment]
I too would like the ability to tell at a glance on the branches and build history whether a build is fully successful or has ignored failures, so I can know if there are lower-priority broken configurations that I need to keep an eye on.

**Comment by dtothefp** [comment]
agreed this would be very helpful, especially if you could combine `allow_failures` with some sort of conditional similar to what has been implemented with `stages`. @BanzaiMan is this even on the radar?

**Comment by gnutix** [comment]
I would love this too! I'd use it to run coding standards checks and such.

**Comment by adrienbernede** [comment]
From [my comment](https://github.com/travis-ci/travis-ci/issues/4730#issuecomment-578325394) in #4730 which was immediately closed, a long time ago.

> A "passed with warnings" would be useful in combination with "allow_failures" IMO.
>
> Allow_failures does not mean "should faild", so I believe there should be a third state "passed with warning" to show, back in the PR, that Travis jobs didn’t run "perfectly" and would use a closer look.
